torchdynamo/variables/tensor.py

```diff
@@ -671,6 +676,17 @@ def from_tensor_variable(cls, tensor_variable, raw_value, need_unwrap=True):
             need_unwrap=need_unwrap,
         )
 
+    def convert_to_constant(self, tx):
+        for graph_arg in tx.output.graphargs:
+            if graph_arg.source is self.source:
+                graph_arg.erase()
+
+        for g in self.guards:
+            if g is self.volatile_guard:
+                g.create_fn = GuardBuilder.CONSTANT_MATCH
```

Comment:
This feels a little sus to me... why repurpose a guard with a new create_fn?

Reply:
When convert to constant, we have to replace the old ```TYPE_MATCH``` guard to ```CONSTANT_MATCH```. We can't replace the whole guard object easily, because it has been propagated to down streams during the symbolic converting.

Reply:
I understand, the question is more around, why not just make a whole new set of guards? If the reason is to remove the guards, we should be explicit, and remove the old guards and add new ones, instead of repurposing.

Reply:
Because the guard has been propagated to many down streams, it's very heavy operation to trigger re-analysis and update the whole set of guards.

Reply:
A whole new set of guards just for this object - why do we need to keep the old guards if the type is changing from tensor to constant? and I have > 1 volatile_guard, does that mean I have > 1 CONSTANT_MATCH for the same object?

Reply:
See this [comment](https://github.com/pytorch/torchdynamo/pull/714#discussion_r939038623)